### PR TITLE
Change detection of FP16 math to follow cutorch setting

### PR DESCRIPTION
**Pull request by borisfom**
As per my cutorch suggestion.
Also. test changes: some tolerances updated for true fp16 
beta fixed to avoid 0 (invariant required in assert is > 0.01)
Volumetric output calculations fixed as per CUDNN team

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -69,8 +69,8 @@ end
 -- TODO: determine if device supports true half and use true half on it
 -- so far use float for half and float, double for double
 local function determineHalfCapability(dev)
-   local prop = cutorch.getDeviceProperties(dev)
-   if prop.major >= 6 or prop.name:find'X1' then
+   -- pick half math option (with possible override) from cutorch
+   if cutorch.hasHalfInstructions then
```
Comment (ngimel) [inline init.lua]: That depends on https://github.com/torch/cutorch/pull/521/files

File: test/test.lua [hunk 2]
```diff
@@ -296,9 +296,11 @@ function cudnntest.VolumetricConvolution()
    local outi = math.random(1,17)
    local outj = math.random(1,17)
    local outk = math.random(1,5)
-   local ini = (outi-1)*si+ki
-   local inj = (outj-1)*sj+kj
-   local ink = (outk-1)*sk+kk
+
+   local ini = outi*si+ki-1
```
Comment (ngimel) [inline test/test.lua]: Why did you change the computation of the input sizes? Previous formulas conformed to nn input size computation.

Comment (borisfom) [inline test/test.lua]: @ngimel : Kevin V. looked at it when we were investigating 3d convolutions crashed. He said the computation was not correct and should be changed like that.

**Comment by borisfom** [comment]
@soumith: FYI, next NVidia Torch package will be shipping with this change.

**Comment by soumith** [comment]
@borisfom by 'this change', do you mean output sizes change? or with this PR?

File: init.lua (hunk 1 above)
Comment (soumith) [inline init.lua]: change this to cutorch.hasHalfInstructions()
See https://github.com/torch/cutorch/commit/073ba88337ff7a584abc6ee11a0f06a0696d8a69

**Comment by borisfom** [comment]
I meant, this PR.

File: init.lua (hunk 1 above)
Comment (borisfom) [inline init.lua]: @soumith : superb! Actually, I believe it should be cutorch.hasFastHalfInstructions()

File: test/test.lua [hunk 3]
```diff
@@ -488,8 +490,8 @@ function cudnntest.SpatialCrossMapLRN_batch()
    local inputSize = math.random(6,9)
    local size = math.random(1,3)*2+1
    local nbfeatures = math.random(3,8)
-   local alpha = math.random(1,100)/100
-   local beta  = math.random(0,100)/100
+   local alpha = math.random(0,100)/100
+   local beta  = math.random(1,100)/100
```
Comment (borisfom) [inline test/test.lua]: This is in fact, a bugfix - as there is an assert(beta>= 0.01) downstream

**Comment by borisfom** [comment]
Superceded by https://github.com/soumith/cudnn.torch/pull/276
